app/routes/index.js

```diff
@@ -0,0 +1,47 @@
+import Backbone from 'backbone';
+import application from '../application';
+import PlaylistsRouter from './playlists';
+
+let Router = Backbone.Router.extend({
+    routes: {
+        'tracks': 'tracks',
+        'upnext': 'upnext',
+        'search?q=:pattern': 'search',
+        '': 'index'
+    },
+
+    initialize() {
+        application.on('start', function() {
+            let playlists = new PlaylistsRouter('playlists');
+        });
+    },
+
+    index() {
+        this.navigate('tracks', { trigger: true });
+    },
+
+    tracks() {
+        application.appState.set('currentPlaylist', application.allTracks);
+    },
+
+    upnext() {
+        application.appState.set('currentPlaylist', application.upNext);
+    },
+
+    search(pattern) {
+        let models = application.allTracks.get('tracks').filter((item) => {
+            let metas = _.defaults({}, item.get('metas'), {
+                artist: '',
+                album: ''
+            });
+            return metas.title.match(new RegExp(pattern, "i"))
```

Comment:
It may be more memory-less to create a single regex pattern to test strings on:
```js
let search = new RegExp(pattern, 'i');
return search.test(metas.title) || search.test(metas.album) || search.test(metas.artist);
```